src/drivers/distance_sensor/lightware_laser_i2c/lightware_laser_i2c.cpp

```diff
@@ -386,6 +400,54 @@ void LightwareLaser::start()
 	ScheduleDelayed(_conversion_interval);
 }
 
+bool LightwareLaser::is_restricted()
+{
+	px4::msg::VehicleStatus vehicle_status;
+
+	if (_vehicle_status_sub.update(&vehicle_status)) {
+		// Check if vehicle type changed
+		if (vehicle_status.vehicle_type != _vehicle_type) {
+			// Transition VTOL -> Fixed Wing
+			if (_vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_ROTARY_WING &&
+			    vehicle_status.vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_FIXED_WING) {
+				_auto_restriction = true;
+			}
+
+			// Transition Fixed Wing -> VTOL
+			else if (_vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_FIXED_WING &&
+				 vehicle_status.vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_ROTARY_WING) {
+				_auto_restriction = false;
+			}
+
+			_vehicle_type = vehicle_status.vehicle_type;
+		}
+	}
+
+	updateParams();
```

Comment:
I don't know the overhead of updating parameters on every iteration. I guess you wanted to save the parameter update subscription. We could just make the setting only apply on reboot 🤷‍♂️

Reply:
Oh no, I'll add the subscription. It needs to be settable by the pilot during flight.